### PR TITLE
Jl/search page

### PR DESCRIPTION
**Pull request by josephli036**

### CODE REVIEW

File: app/controllers/songs_controller.rb [hunk 1]
```diff
@@ -1,10 +1,25 @@
...
 
   def edit
   end
+  
+  def search
+    @results = true
+    if params[:search].nil?
+      @songs = Song.all
+    else
+      @songs = Song.search params[:search]
```
Comment (patrickf3139) [inline app/controllers/songs_controller.rb]: Query logic shouldn't be contained in the model generally. After all, it makes sense models wouldn't know how to retrieve themselves.

File: app/controllers/songs_controller.rb [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
+      if @songs.nil? or @songs.none?
```
Comment (patrickf3139) [inline app/controllers/songs_controller.rb]: Don't use `or`, use `||`. http://devblog.avdi.org/2010/08/02/using-and-and-or-in-ruby/

File: app/controllers/songs_controller.rb [hunk 3]
```diff
@@ -1,10 +1,25 @@
...
   def new
   end
 
   def edit
   end
+  
+  def search
+    @results = true
+    if params[:search].nil?
+      @songs = Song.all
```
Comment (patrickf3139) [inline app/controllers/songs_controller.rb]: Might want to paginate so when we have 1000 songs, they don't all get loaded. Check out https://github.com/mislav/will_paginate

File: test/models/song_test.rb [hunk 4]
```diff
@@ -33,4 +33,16 @@ class SongTest < ActiveSupport::TestCase
...
+  
+  test "search returns correct records" do
+    results = Song.search('God')
+    correct_results = songs(:God_be_praised)
+    assert_equal results.first, correct_results
+    assert results.one?
+  end
+  
+  test "search returns no records if no results" do
+    results = Song.search('banana')
```
Comment (patrickf3139) [inline test/models/song_test.rb]: Hmmm I can see banana being in a song. Haha who knows? How about something that really could never exist like "e211d319d9b52c9f9215656db9c33deb".

**Comment by patrickf3139** [comment]
I've been thinking about how to perform our queries. It turns out MySQL has some full-text capabilities built in (https://dev.mysql.com/doc/refman/5.5/en/fulltext-natural-language.html), so we won't need a separate server after all. 

I think this'll definitely take some complicated logic, or at least our search will be a WIP for the upcoming weeks. Give this, let's abstract away the query logic and put it in a library. @josephli036 , can you put the logic in a file called song_finder.rb under app/finders?

File: app/controllers/songs_controller.rb [hunk 5]
```diff
@@ -1,10 +1,25 @@
 class SongsController < ApplicationController
   def index
+    @songs = Song.all
```
Comment (leethomas) [inline app/controllers/songs_controller.rb]: This is okay for now, but `all` actually loads every record in the database into memory. I'm working on adding datatables tonight, which will paginates the query for us.

Comment (patrickf3139) [inline app/controllers/songs_controller.rb]: Wait a minute: why do we have both the index and search actions? They can be just one method. Let's just call it index so move the search code into index.

Also, `Song.all` can probably be `Song.order(:name)`.

**Pull request closed without merging**